go/test/endtoend/vreplication/vreplication_test.go

```diff
@@ -173,7 +174,19 @@ func TestVReplicationDDLHandling(t *testing.T) {
 	waitForQueryResult(t, vtgateConn, targetKs, checkColQueryTarget, "[[INT64(0)]]")
 	// Confirm new col does exist on source
 	waitForQueryResult(t, vtgateConn, sourceKs, checkColQuerySource, "[[INT64(1)]]")
-	moveTablesAction(t, "Cancel", defaultCellName, workflow, sourceKs, targetKs, table)
+	// Also test Cancel --keep_routing_rules
+	moveTablesAction(t, "Cancel", defaultCellName, workflow, sourceKs, targetKs, table, "--keep_routing_rules")
+	// Confirm that the routing rules were NOT cleared
+	rr, err := vc.VtctldClient.ExecuteCommandWithOutput("GetRoutingRules")
+	require.NoError(t, err)
+	require.Greater(t, len(gjson.Get(rr, "rules").Array()), 0)
```

Comment:
In https://github.com/vitessio/vitess/pull/12861/files#r1160655110 I've found `jsondiff` to give much easier to understand output.

In general, we should really standardize on one and remove all the other json libraries we're using for diffing :smile:.

Reply:
Ah, that also uses `gjson` (see https://github.com/wI2L/jsondiff/blob/master/go.mod), so that's even with this cleanup idea still the one we'd like to use. Nvm me then here :smile:.